### PR TITLE
implementation guideline

### PR DESCRIPTION
**Pull request by dataders**
Resolves #
Depends on: #471

### Description

mostly worked on the python class tuff

- [ ] applied state
- [ ] relation-specific drop and rename
- [ ] modify materializations according to above
- [x] (not all the way done yet) Python class configs
- [ ] new MV tests

### Checklist

- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-databricks next" section.

### CODE REVIEW

**Comment by benc-db** [comment]
Please see tests failing re: hologram removal.

**Comment by benc-db** [comment]
@dataders is this ready for me to take over or do you still plan on another commit?

**Comment by dataders** [comment]
> @dataders is this ready for me to take over or do you still plan on another commit?

not even close. 🥲  i'll be working on this more today and tomorrow. stay tuned.

also `dbt-spark 1.6.0b2` (which includes the hologram work https://github.com/dbt-labs/dbt-spark/pull/880) is being cut as we speak. this'll obviate my most recent commit

File: tests/functional/adapter/materialized_view_tests/utils.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        case
+            when definition ilike '%create materialized view%'
+            then 'materialized_view'
+            else 'view'
+        end as relation_type
+    from pg_views
+    where schemaname = '{relation.schema}'
+    and viewname = '{relation.identifier}'
+    """
+    results = project.run_sql(sql, fetch="all")
```
Comment (dataders) [inline tests/functional/adapter/materialized_view_tests/utils.py:28]: @mikealfare how can I change this to use `list_relations_without_caching()` instead? in my mind that's where this logic is already standardized.

in the case of dbt-databricks, it all the logic is in Python ( [`DatabricksAdapter.get_relations_without_caching()`](https://github.com/dataders/dbt-databricks/blob/539a40f2e9035386b8e96d341908b4795b5f6d43/dbt/adapters/databricks/impl.py#L221-L297). I'd rather not duplicate/reimplement it

Comment (dataders) [reply tests/functional/adapter/materialized_view_tests/utils.py:28]: in the meantime, i suppose this is a decent woraround
```sql
    select
        'table' as relation_type
    from information_schema.tables
    where schemaname = '{relation.schema}'
    and tablename = '{relation.identifier}'
    union all
    select
        case when
            is_materialized = TRUE then 'materialized_view'
            else 'view'
        end as relation_type
    from information_schema.views
    where schemaname = '{relation.schema}'
    and viewname = '{relation.identifier}'
```

Comment (benc-db) [reply tests/functional/adapter/materialized_view_tests/utils.py:28]: So long as its only used with the UC cluster...information schema isn't available on the non-uc cluster.

**Comment by benc-db** [comment]
@dataders status update?  At any point feel free to hand off and give me instructions on what is remaining :).

**Comment by dataders** [comment]
> @dataders status update? At any point feel free to hand off and give me instructions on what is remaining :).

you @benc-db ask and you shall receive :) #473

**Comment by benc-db** [comment]
> > @dataders status update? At any point feel free to hand off and give me instructions on what is remaining :).
> 
> you @benc-db ask and you shall receive :) #473

Ok, taking it now.

**Comment by dataders** [comment]
@benc-db, after discussing internally, we're dropping this MV refactor work as a requirement for `1.7.0rc1` or even `1.7.0`, but since you already have your hands on it, I wanted to show the next step. I was afraid to push directly to the branch in this PR, so I made https://github.com/dataders/dbt-databricks/pull/1 that you can pull in if you like. #477 gives more context

**Pull request closed without merging**